consensus/reactor.go

```diff
@@ -94,11 +101,57 @@ func WithGossipDataEnabled(enabled bool) ReactorOption {
 	}
 }
 
+// WithConfig sets the consensus config for the reactor
+func WithConfig(config *cfg.ConsensusConfig) ReactorOption {
+	return func(conR *Reactor) {
+		conR.config = config
+	}
+}
+
 // IsGossipDataEnabled returns whether the gossipDataRoutine should run
 func (conR *Reactor) IsGossipDataEnabled() bool {
 	return conR.gossipDataEnabled.Load()
 }
 
+// checkAndDisableOldPropagationRoutine checks if conditions are met to disable
+// the old propagation routine (gossipDataRoutine) when app version >= 5.
+// Conditions:
+// 1. App version >= 5
+// 2. New propagation reactor is not disabled 
+// 3. Old routine is not already disabled
+func (conR *Reactor) checkAndDisableOldPropagationRoutine() {
+	// Only check if we have config and old routine is currently enabled
+	if conR.config == nil || !conR.IsGossipDataEnabled() {
+		return
+	}
+
+	// Check if new propagation reactor is not disabled
+	if conR.config.DisablePropagationReactor {
+		return
+	}
+
+	// Get the current consensus state to check app version
+	conR.conS.mtx.RLock()
+	state := conR.conS.state
+	appVersion := state.ConsensusParams.Version.App
+	conR.conS.mtx.RUnlock()
```

Comment:
let's try to use the app version after it is passed back from the application if possible